### PR TITLE
Editor server cleanup

### PR DESCRIPTION
**Pull request by LordSputnik**
### Problem
The code for handling editor route requests currently uses long chains of promises, which can be hard to follow and reason about. The current use of promises also enforces unnecessary serialization of some database operations. Code is duplicated in several places, and some minor bugs are present.

### Solution
This PR mainly re-writes the code using promise chains to use ES6 async/await to express asynchronous code in a flat, linear format. It also refactors some code to pull out commonly used data-fetching snippets into separate functions, which could be migrated to the bookbrainz-data-js package in future. Some other minor code quality improvements are also implemented.

### Areas of Impact
Impacts two files:
* `src/server/routes/editor.js`
* `src/server/helpers/handler.js`

The change to handler.js impacts how responses for POST requests are generated, but shouldn't change functionality. The changes to editor.js affect the following routes, but again, shouldn't change any functionality:

* `/editor/edit`
* `/editor/edit/handler`
* `/editor/:id`
* `/editor/:id/revisions`
* `/editor/:id/revisions/revisions`
* `/editor/:id/achievements`
* `/editor/:id/collections`
* `/editor/:id/collections`

### CODE REVIEW

**Comment by MonkeyDo** [comment]
Thanks, that looks a lot cleaner !
I'll do a proper review, but I wanted to let you know first to save you some bug hunting that the failing travis test suite is due to a newly release Node LTS version and a bug in the node-postgres package, which I'm working to fix as I write this.

File: src/server/routes/editor.js [hunk 1]
```diff
@@ -357,170 +354,180 @@ router.get('/:id/revisions', async (req, res, next) => {
...
+	async function runAsync() {
+		if (!isCurrentUser(userId, req.user)) {
+			throw new Error('Not authenticated');
+		}
+
+		// Presumably, this is testing that the provided editor ID is valid
+		// prior to attempting to update the ranks. Possibly not needed since
+		// the user must exist if authenticated?
+		await new Editor({id: userId})
+			.fetch({require: true});
```
Comment (MonkeyDo) [inline src/server/routes/editor.js:489]: I agree, I can't see any way for the user not to exist, and the `auth.isAuthenticated` middleware ensure that.
I think it can be removed.

**Review by MonkeyDo — CHANGES_REQUESTED** [review]
Nicely done, that's a great refactor, and sorely needed too! Thanks a lot for tackling this.

I've tested everything I could think of, everything is working fine apart from on Promise.all call.
I've paced through all the code and found nothing else amiss.

File: src/server/routes/editor.js [hunk 2]
```diff
@@ -241,85 +231,92 @@ export async function getEditorActivity(editorId, startDate, Revision, endDate =
...
-		const orderedRevisions = await getOrderedRevisionForEditorPage(from, size + 1, req);
-		const {newResultsArray, nextEnabled} = utils.getNextEnabledAndResultsArray(orderedRevisions, size);
-		const editor = await new Editor({id: req.params.id}).fetch();
-		const editorJSON = await getEditorTitleJSON(editor.toJSON(), TitleUnlock);
+		const orderedRevisionsPromise =
+			getOrderedRevisionForEditorPage(from, size + 1, req);
+		const editorJSONPromise = getIdEditorJSONPromise(req.params.id, req);
+
+		const [orderedRevisions, editorJSON] =
+			await Promise.all(orderedRevisionsPromise, editorJSONPromise);
```
[the remark this answers is not in the document]
Comment (LordSputnik) [reply src/server/routes/editor.js:316]: Good spot, thanks!

[the remark this answers is not in the document]
Comment (LordSputnik) [reply src/server/routes/editor.js]: Fixed in 0d893fd

File: src/server/routes/editor.js [hunk 3]
```diff
@@ -44,166 +44,154 @@ import target from '../templates/target';
...
-			editorForES.aliasSet = {
+		const editor = await Editor
+			.forge({id: parseInt(req.user.id, 10)})
+			.fetch({require: true})
+			.catch(Editor.NotFoundError, () => {
+				throw new error.NotFoundError('Editor not found', req);
+			});
+
+		// Modify the user to match the updates from the form
+		const titleID = _.get(req.body, 'title', null) || null;
```
[the remark this answers is not in the document]
Comment (LordSputnik) [reply src/server/routes/editor.js]: Removed in b2ec79a

File: src/server/routes/editor.js (hunk 1 above)
Comment (LordSputnik) [reply src/server/routes/editor.js]: Removed in b2ec79a

**Comment by LordSputnik** [comment]
Fixed issues and rebased, thanks for the thorough testing and review 😃

**Review by MonkeyDo — APPROVED** [review]
Thanks a lot !
Really nice to see some code quality improvements PRs, and even better to see PRs from you ! :)
